p2p/src/connection/outgoing/p2p_connection_outgoing_reducer.rs

```diff
@@ -509,19 +529,30 @@ impl P2pConnectionOutgoingState {
                     return Ok(());
                 }
 
+                let Some(peer_state) = p2p_state.peers.get_mut(&peer_id) else {
+                    bug_condition!("Outgoing peer state not found for: {}", peer_id);
+                    return Ok(());
+                };
+                let callback = peer_state.on_connect_success.take();
```

Comment:
Related to the content I left above, it would be good to have a comment explaining this a bit, because it is not immediately obvious